### DISCUSSION TITLE
UMEP with QGIS docker image

### DISCUSSION THREAD

**Discussion by torm89** [UMEP-dev/UMEP#493] [Python]
Hi,

I want to run the UMEP plugin in docker container using the [official QGIS docker image](https://hub.docker.com/r/qgis/qgis). 

There is a problem with import of `ProcessingUMEPProvider`. When I'm trying to import it I got
 ```
This plugin does not support propagateSizeHints()
``` 
warning and script processing stops (screen becomes frozen).
 
Does someone know how to resolve that issue?

**Comment by biglimp** [comment]
I nvere tried to access UMEP like this so I dont know. I dont think this propagateSizeHints() is something related to UMEP itself but to e.g. Qt5 or something. Have you tried to add any other third party processing plugin? There seem to be other who have issues with this function also if you google it.

**Reply by torm89** [reply]
I haven't try to add other pluging (except processing) in python script but I was tring to enable plugins with `qgis_process` command. 

[OpenTripPlanner](https://plugins.qgis.org/plugins/processing_saga_nextgen/) can be enabled with this command.

![UMEP02](https://user-images.githubusercontent.com/5539323/213455336-b437fc54-d5b8-467d-8034-fc256dc5c2eb.png)

For UMEP I got the same error

![UMEP01](https://user-images.githubusercontent.com/5539323/213455510-f17ea690-c88b-4706-a84e-8936bdea4275.png)

So, I suppose that the error is related to UMEP plugin.

**Reply by biglimp** [reply]
Is not otb a core plugin? Try e.g. with QuickOSM. UMEP does not use propagateSizeHints() so it must be coming from some lib that we are importing.

**Reply by torm89** [reply]
I'm able to enable QuickOSM plugin with `qgis_process`

![image](https://user-images.githubusercontent.com/5539323/213461803-8faf8fd4-f43f-447c-b255-41addbdd20d3.png)

**Reply by biglimp** [reply]
Ok. Then try to find out what library that make use of that function and maybe we can fix it.

**Reply by torm89** [reply]
OK. With a bit of luck I've found that the problem is caused by code in `suewsmodel/__init__.py` file. 
```python
# -*- coding: utf-8 -*-
# __author__ = 'xlinfr'

from qgis.PyQt.QtWidgets import QMessageBox
from .supy_installer import setup_supy
from qgis.core import Qgis, QgsMessageLog
# we can specify a version if needed
try: 
    import supy as sp
    from supy import __version__ as ver_supy
    QgsMessageLog.logMessage("UMEP - SuPy Version installed: " + ver_supy, level=Qgis.Info)
except:
    if QMessageBox.question(None, "Supy and related dependencies not installed", 
              "Do you want UMEP to automatically install missing python modules? \r\n"
              "QGIS will be non-responsive for a couple of minutes.", 
               QMessageBox.Ok | QMessageBox.Cancel) == QMessageBox.Ok:
        setup_supy(ver=None)
        try:
            setup_supy(ver=None)
            QMessageBox.information(None, "Packages successfully installed", 
                                    "To make all parts of the plugin work it is recommended to restart your QGIS-session.")
        except Exception as e:
            QMessageBox.information(None, "An error occurred",
                                    "Packages not installed. report any errors to https://github.com/UMEP-dev/UMEP/issues")
    else:
        QMessageBox.information(None,
                                "Information", "Packages not installed. Some UMEP tools will not be fully operational.")
# setup_supy(ver='2020.1.23')
```

I didn't have `supy` installed, so the program was trying to show me a message box and ask about installation of "missing python modules".  Because I was in console and had `QT_QPA_PLATFORM=offscreen` this information should be printed in console, simply by `print` function.
